test/integration/operator_run_bundle_test.go

```diff
@@ -0,0 +1,235 @@
+		if !runBundleImplemented {
+			Skip("Run bundle not yet implemented")
+		}
+
+		cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+		err := cfg.Load()
+		Expect(err).NotTo(HaveOccurred())
+
+		i := bundle.NewInstall(cfg)
+		i.BundleImage = tc.ImageName
+		i.InstallMode.InstallModeType = v1alpha1.InstallModeTypeOwnNamespace
+		i.InstallMode.TargetNamespaces = []string{tc.Kubectl.Namespace}
+
+		err = doInstall(i)
+		Expect(err).NotTo(HaveOccurred())
+
+		uninstall(kubeconfigPath, tc)
+	})
+
+	It("Run Bundle - watching the specified namesapce", func() {
+		if !runBundleImplemented {
+			Skip("Run bundle not yet implemented")
+		}
+
+		cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath, Namespace: tc.Kubectl.Namespace}
+		err := cfg.Load()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("create a namespace to watch")
+		_, err = tc.Kubectl.Command("create", "namespace", "bar")
+		Expect(err).NotTo(HaveOccurred())
+
+		i := bundle.NewInstall(cfg)
+		i.BundleImage = tc.ImageName
+		i.InstallMode.InstallModeType = v1alpha1.InstallModeTypeSingleNamespace
+		i.InstallMode.TargetNamespaces = []string{tc.Kubectl.Namespace, "bar"}
+
+		err = doInstall(i)
+		Expect(err).NotTo(HaveOccurred())
+
+		uninstall(kubeconfigPath, tc)
+	})
+
+})
+
+// createOperator scaffolds a new test operator, generates bundle and pushes it to a remote
+// repository. The generated csv by default supports OwnNamespace, SingleNamespace and AllNamespaces.
+func createOperator(tc testutils.TestContext, projectName string) {
+	By("initializing a project")
+	err := tc.Init(
+		"--project-version", "3-alpha",
+		"--repo", path.Join("github.com", "example", projectName),
+		"--domain", tc.Domain,
+		"--fetch-deps=false")
+	Expect(err).Should(Succeed())
+
+	By("creating an API definition")
+	err = tc.CreateAPI(
+		"--group", tc.Group,
+		"--version", tc.Version,
+		"--kind", tc.Kind,
+		"--namespaced",
+		"--resource",
+		"--controller",
+		"--make=false")
+	Expect(err).Should(Succeed())
+
+	By("implementing the API")
+	Expect(kbtestutils.InsertCode(
+		filepath.Join(tc.Dir, "api", tc.Version, fmt.Sprintf("%s_types.go", strings.ToLower(tc.Kind))),
+		fmt.Sprintf(`type %sSpec struct {
+`, tc.Kind),
+		`	// +optional
+	Count int `+"`"+`json:"count,omitempty"`+"`"+`
+`)).Should(Succeed())
+
+	By("generating the operator bundle")
+	// Turn off interactive prompts for all generation tasks.
+	replace := "operator-sdk generate kustomize manifests"
+	testutils.ReplaceInFile(filepath.Join(tc.Dir, "Makefile"), replace, replace+" --interactive=false")
+	err = tc.Make("bundle", "IMG="+tc.ImageName)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("building the operator bundle image")
+	// Use the existing image tag but with a "-bundle" suffix.
+	err = tc.Make("bundle-build", "BUNDLE_IMG="+tc.ImageName)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("push the bundle to a remote repository")
+	err = tc.Make("docker-push", "IMG="+tc.ImageName)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
```suggestion
```
we should not use the docker-push here. The tests are executed using Kind. We build the images and then we load them. See; 

https://github.com/operator-framework/operator-sdk/blob/master/test/e2e-ansible/e2e_ansible_olm_test.go#L62-L67